xarray/core/dataarray.py

```diff
@@ -840,6 +840,99 @@ def swap_dims(self, dims_dict):
         ds = self._to_temp_dataset().swap_dims(dims_dict)
         return self._from_temp_dataset(ds)
 
+    def expand_dims(self, dim=None, axis=None):
+        """Return a new object with an additional axis (or axes) inserted at the
+        corresponding position in the array shape.
+
+        If dim is already a scalar coordinate, it will be promoted to a 1D
+        coordinate consisting of a single value.
+
+        Parameters
+        ----------
+        dim : str or sequence of str, or None.
+            Dimensions to include on the new variable.
+            dimensions are inserted with length 1.
+            If None is passed, dimension labels are automatically generated,
```

Comment:
I would rather not support automatic dimension naming anywhere other than the DataArray constructor. It's error prone if `array.expand_dims()` usually makes an array with a first dimension `dim_0` and but sometimes calls it something else.

It's OK for DataArray constructors because (a) it mostly reminds users that they need to supply names and (b) usually every name is given a default value, so we don't need to worry about conflicts.